executor/cook/io_helper.py

```diff
@@ -73,12 +77,16 @@ def print_err(string_data, flush=False, newline=True):
     -------
     Nothing.
     """
+    stderr_buffer = sys.stderr.buffer
     with __stderr_lock__:
-        sys.stderr.write(string_data)
+        if isinstance(data, str):
+            stderr_buffer.write(data.encode())
+        else:
+            stderr_buffer.write(data)
         if newline:
-            sys.stderr.write(os.linesep)
+            stderr_buffer.write(os.linesep)
         if flush:
-            sys.stderr.flush()
+            stderr_buffer.flush()
```

Comment:
I still don't like the fact that this is repeated (once for `stdout` and once for `stderr`), especially since they're not even adjacent. Even though they're pretty simple, this still makes it harder to maintain since the code is in two places. I think the fact that the `if newline` branches don't match right now is a prime example of this issue.

Reply:
Created `print_to_buffer` helper function. I did not understand the `newline` branches not matching comment.